packages/i18n/src/t/translation-binding.ts

```
@@ -55,7 +55,7 @@ export class TranslationBinding implements IPartialConnectableBinding {
   public parametersExpr?: IsExpression;
   private readonly i18n: I18N;
   private readonly contentAttributes: readonly string[] = contentAttributes;
-  private keyExpression!: string;
+  private keyExpression!: string | undefined | null;
```

I have no idea why the keyExpression was created with a !, guess to suppress some TS issues. Anyways when `t.bind="whatever"` is bound to a non-string, pretty much anything can land in there

---

The `!` assertion is to avoid the truthy assertion for future (anything after `$bind`). I think we should not allows binding `0` at all. If the key expression does not evaluate to string (refer `this.keyExpression = this.expr.evaluate(flags, scope, this.locator, part) as string;` in `$bind`), we should simply throw an error. With this place, you can have this 
```typescript
const results = this.i18n.evaluate(this.keyExpression ?? '', this.translationParameters);
```
instead of 
```typescript
const keyExpressionSanitized = this.keyExpression === null || this.keyExpression === undefined
  ? ""
  : this.keyExpression;
const results = this.i18n.evaluate(keyExpressionSanitized, this.translationParameters);
```

IMO a non-string i18n key does not make any sense. The test is currently failing because `0.split` does not exist.